apps/vs-code-designer/src/app/commands/createNewCodeProject/cloudToLocal.ts

```diff
@@ -21,14 +23,99 @@ import { setWorkspaceName } from './CodeProjectBase/SetWorkspaceName';
 import { AzureWizard } from '@microsoft/vscode-azext-utils';
 import type { IActionContext } from '@microsoft/vscode-azext-utils';
 import { latestGAVersion, OpenBehavior } from '@microsoft/vscode-extension-logic-apps';
-import type { ICreateFunctionOptions, IFunctionWizardContext, ProjectLanguage } from '@microsoft/vscode-extension-logic-apps';
+import type {
+  ConnectionReferenceModel,
+  ICreateFunctionOptions,
+  IFunctionWizardContext,
+  ProjectLanguage,
+} from '@microsoft/vscode-extension-logic-apps';
 import { window } from 'vscode';
 import * as path from 'path';
-import * as util from 'util';
 import AdmZip = require('adm-zip');
 import * as fs from 'fs';
+
+// Constants
 const openFolder = true;
+const DELIMITER = '/';
+const SUBSCRIPTION_INDEX = 2;
+const MANAGED_API_LOCATION_INDEX = 6;
+const MANAGED_CONNECTION_RESOURCE_GROUP_INDEX = 4;
+
+// Function to create an AdmZip instance
+function createAdmZipInstance(zipFilePath: string) {
+  return new AdmZip(zipFilePath);
+}
+
+// Function to get zip entries
+function getZipEntries(zipFilePath: string) {
+  const zip = createAdmZipInstance(zipFilePath);
+  return zip.getEntries();
+}
+
+// Function to deep merge objects
+function deepMergeObjects(target: any, source: any): any {
+  const result = { ...target };
+  for (const key of Object.keys(source)) {
+    if (source[key] instanceof Object && key in target) {
+      result[key] = deepMergeObjects(target[key], source[key]);
+    } else {
+      result[key] = source[key];
+    }
+  }
+  return result;
+}
 
+// Function to extract connection details
+function extractConnectionDetails(connection: ConnectionReferenceModel): any {
```

Comment:
Could you please add unit test for this and `changeAuthTypeToRaw` functions. You can see some examples here apps/vs-code-designer/src/app/utils/codeless/__test__/parameterizer.test.ts

Reply:
+1 here

Reply:
Added two unit test for this in new pr